viewer/src/components/Commands.vue

```diff
@@ -685,19 +700,22 @@ export default class Commands extends Vue {
 
   mounted() {
     this.keyListener = (e) => {
-      if (e.key == "Escape") {
-        if (this.commandChain.length > 0) {
-          this.back();
-        } else if (!this.$store.state.gaiaViewer.data.newTurn) {
-          this.undo();
-        }
+      if (e.key == "Escape" && this.$store.getters["gaiaViewer/canUndo"]) {
+        this.undo();
       }
     };
     window.addEventListener("keydown", this.keyListener);
+    this.undoListener = this.$store.subscribeAction(({ type, payload }) => {
+      if (type === "gaiaViewer/undo") {
+        this.back();
+      }
+    });
```

Comment:
For those, instead of storing them in a property of the class, you can do:

```
const keyListener = ...;
this.$on("hook:beforeDestroy", () => window.removeEventListener(...));

const undoUnlisten = ...
this.$on("hook:beforeDestroy", undoUnlisten);
```

This:
- puts all the code in one place
- no need to add properties on the class

For `subcriptions`, given it's reset / accessed during the lifetime of the component, I'm not sure it's possible.